### PR TITLE
core: refactor thread info 

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

This PR untangles core and ps a bit, by turning some of ps' direct accesses to thread_t fields to core API calls.
There are some left (mostly regarding stack defines), will do those in a later PR.

I consider this a minor refactor, so IMO no need for two ACKs.

### Testing procedure

ps output is widely tested on CI, so a successful run should suffice.

### CODE REVIEW

File: core/thread.c [hunk 1]
```diff
@@ -318,3 +318,31 @@ kernel_pid_t thread_create(char *stack, int stacksize, uint8_t priority,
...
+    [STATUS_COND_BLOCKED] = "bl cond",
+    [STATUS_RUNNING] = "running",
+    [STATUS_PENDING] = "pending",
+};
+
+#define STATE_NAME_UNKNOWN "unknown"
+
+const char *thread_state_to_string(thread_status_t state)
+{
+    const char *name = state_names[state] ? state_names[state] : NULL;
```
Comment (benpicco) [inline core/thread.c:342]: shouldn't that be 
```suggestion
    const char *name = state < state_names ? state_names[state] : NULL;
```

Comment (kaspar030) [reply core/thread.c:342]: (this was copied from ps.c unchanged) no, state is an enum, and state_names an array that gets indexed.
But now that I'm taking a closer look this is essentially:

```
if state_names[state] == NULL:
   name = NULL;
else:
  name = state_names[state]
```

So the assignment could just be ```const char *name = state_names[name];```, right?

Or was that check meant for validity of "state"? In that case it must be ```const char *name = state < STATUS_NUMOF ? state_names[state] : NULL;```.

Comment (kaspar030) [reply core/thread.c:342]: Anyhow, I'd like to keep any changes other than the refactoring out of this PR.

**Review by benpicco — APPROVED** [review]

**Comment by kaspar030** [comment]
Thanks!

**Pull request merged**